documentation/course/e2/README.md

```diff
@@ -269,44 +299,125 @@ TODO @mlodybercik: porównanie z on-premise i najlepiej innymi chmurami
     <th>Wady</th>
   </tr>
   <tr>
-    <th>Rozwiązanie 1</th>
+    <th><i>on-premise</i></th>
     <td>
       <ul>
-        <li>Zaleta 1</li>
+        <li>Pełna kontrola nad danymi</li>
+        <li>Brak zależności od dostawcy usług</li>
+        <li>Niski koszt przy wysokim obciążeniu</li>
+        <li>Niskie opóźnienia</li>
```

Comment:
I'm not sure about that but maybe it's true. In what sense? Because I guess it's by using the cloud that you can easily have servers in multiple locations and you can be as close to the customer as possible.

Reply:
On-premise basicly means that you have your own servers in your office building.  Closest AWS datacenter from Poland is in Frankfurt and Stockholm. By having servers in your office you basicly cut the distance data has to travel. By having our own datacenter in Wrocław and hosting this app for residents of Wrocław you cut latencies far more than half.